### PR TITLE
feat: introduce TestFramework class to evaluate child executePipeline…

### PR DESCRIPTION
**Pull request by arjendev**
This PR introduces:
- The TestFramework class which will be responsible for managing repositories and evaluation of pipelines, linkedServices, datasets and triggers. 
- The DataFactoryRepository class which will parse the entire data-factory folder.
- Evaluation of child-pipelines (ExecutePipelineActivity) if enabled in the TestFramework constructor.

### CODE REVIEW

File: src/AzureDataFactory.TestingFramework.Example/BatchJob/BatchJobUnitTests.cs [hunk 1]
```diff
@@ -66,8 +66,8 @@ public void SetManagerApplicationPackagePathTest()
         // Arrange
         var activity = _pipeline.GetActivityByName("Set ManagerApplicationPackagePath") as SetVariableActivity;
         var variable = new PipelineRunVariable<string>("ManagerApplicationPackagePath");
-        _state.Parameters.Add(new RunParameter<string>(ParameterType.Parameter, "ManagerApplicationPackageName", "managerworkload"));
-        _state.Parameters.Add(new RunParameter<string>(ParameterType.Parameter, "ManagerApplicationPackageVersion", "0.13.2"));
```
Comment (ydaponte) [inline src/AzureDataFactory.TestingFramework.Example/BatchJob/BatchJobUnitTests.cs:70]: Could we create a unique variable that holds the versions of the ManagerApplicationPackageVersions for the different package names. The idea is if the version changes it doesn't need to be changed on all different possible  places is being used.

File: src/AzureDataFactory.TestingFramework/AzureDataFactory.TestingFramework.csproj [hunk 2]
```diff
@@ -11,12 +11,17 @@
...
+    <ItemGroup>
+        <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
+            <_Parameter1>AzureDataFactory.TestingFramework.Tests</_Parameter1>
+        </AssemblyAttribute>
+    </ItemGroup>
+    
     <PropertyGroup>
         <PackageId>AzureDataFactory.TestingFramework</PackageId>
-        <PackageVersion>0.1.6-alpha</PackageVersion>
+        <PackageVersion>0.1.7-alpha</PackageVersion>
```
Comment (ydaponte) [inline src/AzureDataFactory.TestingFramework/AzureDataFactory.TestingFramework.csproj:24]: We need to document the version evolution in the main ReadMe and document the differences (weather if its the ADF underlying version that changes and/or the framework itself)

File: src/AzureDataFactory.TestingFramework/Models/Pipelines/Pipeline.cs [hunk 3]
```diff
@@ -23,13 +23,12 @@ public PipelineActivity GetActivityByName(string activityName)
...
-    /// Evaluates the pipeline with the provided parameters. The order of activity execution is simulated based on the dependencies. Any expression part of the activity is evaluated based on the state of the pipeline.
+    /// Validates whether the list of parameters are complete and not duplicate.
     /// </summary>
-    /// <param name="parameters">The global and regular parameters to be used for evaluating expressions.</param>
-    /// <returns></returns>
-    /// <exception cref="PipelineParameterNotProvidedException">Thrown if a required pipeline parameter is not required</exception>
-    /// <exception cref="PipelineDuplicateParameterProvidedException">Thrown if a pipeline parameter is provided more than once</exception>
-    public IEnumerable<PipelineActivity> Evaluate(List<IRunParameter> parameters)
+    /// <param name="parameters"></param>
+    /// <exception cref="PipelineParameterNotProvidedException"></exception>
```
Comment (ydaponte) [inline src/AzureDataFactory.TestingFramework/Models/Pipelines/Pipeline.cs:29]: The description of the exception was deleted from the previous version - not sure if it was intentional as the name is self-explanatory, but there are other methods that are including the descriptions. We should just keep the consistency on how do we do it.

**Review by ydaponte — APPROVED** [review]
Just minor comments, not blocking the PR. You can go ahead and merge.
Just in the PR itself, when you create the PR link it to the issue you are solving. In  this case add something like " PR closes issue #7 ". We might need to create a PR template with a checklist.

File: src/AzureDataFactory.TestingFramework/AzureDataFactory.TestingFramework.csproj (hunk 2 above)
Comment (arjendev) [reply src/AzureDataFactory.TestingFramework/AzureDataFactory.TestingFramework.csproj:24]: Great idea, I'll add a separate issue to introduce a changelog.

File: src/AzureDataFactory.TestingFramework/Models/Pipelines/Pipeline.cs (hunk 3 above)
Comment (arjendev) [reply src/AzureDataFactory.TestingFramework/Models/Pipelines/Pipeline.cs]: Nice catch! Fixed.

File: src/AzureDataFactory.TestingFramework.Example/BatchJob/BatchJobUnitTests.cs (hunk 1 above)
Comment (arjendev) [reply src/AzureDataFactory.TestingFramework.Example/BatchJob/BatchJobUnitTests.cs:70]: It's just an example value and not being used by anything. Will leave this for now :)

**Comment by arjendev** [comment]
> Just minor comments, not blocking the PR. You can go ahead and merge. Just in the PR itself, when you create the PR link it to the issue you are solving. In this case add something like " PR closes issue #7 ". We might need to create a PR template with a checklist.

I referenced the issue in the PR now (not in text but as part of the PR).

**Review by ydaponte — APPROVED** [review]
LGTM!! Thanks!

**Pull request merged**